src/backend/executor/execSRF.c

```diff
@@ -972,3 +978,38 @@ tupledesc_match(TupleDesc dst_tupdesc, TupleDesc src_tupdesc)
 							   i + 1)));
 	}
 }
+
+
+void
+ExecSquelchFunctionResultSet(SetExprState *fcache,
+							 ExprContext *econtext,
+							 MemoryContext *argcontext)
+{
+	FunctionCallInfo fcinfo = fcache->fcinfo;
+
+	/*
+	 * if fcache->shutdown_reg  is set to true, the function is known to be
+	 * called at least once. Otherwise it was not called and it has no need to
+	 * release resources.
+	 */
+	if (SRF_IS_FIRSTCALL() || !fcache->shutdown_reg || !fcache->isSquelchSupported)
+	{
+		return;
+	}
+
+	/* Prepare a resultinfo node for communication. */
+	ReturnSetInfo rsinfo = {
+		.type = T_ReturnSetInfo,
+		.econtext = econtext,
+		.expectedDesc = fcache->funcResultDesc,
+		.allowedModes = (int) (SFRM_ValuePerCall | SFRM_Materialize | SFRM_Squelch),
```

Comment:
Why do we need the `SFRM_ValuePerCall` and `SFRM_Materialize` flags?

Reply:
Perhaps we don't, will check work without them.